In [3]:
#Imports
import zipfile
import os
import pandas as pd


In [ ]:
# Unzip the dataset
def unzip_csv_files(zip_path, extract_to):
    if not os.path.exists(extract_to):
        os.makedirs(extract_to)

    with zipfile.ZipFile(zip_path, 'r') as zip_ref:
        csv_files = [file for file in zip_ref.namelist() if file.endswith('.csv')]

        for file in csv_files:
            zip_ref.extract(file, extract_to)

zip_file_path = "/content/drive/MyDrive/mammal_species/mammal_species.zip"
output_directory = "/content/mammal_species"

unzip_csv_files(zip_file_path, output_directory)

In [4]:
# Main csv file with all the species
df_main = pd.read_csv('/content/mammal_species/MDD_v1.10_6615species.csv')
df_main.head()

sciName       id  phylosort  \
0  Ornithorhynchus_anatinus  1000001          1   
1    Tachyglossus_aculeatus  1000002          1   
2   Zaglossus_attenboroughi  1000003          1   
3         Zaglossus_bartoni  1000004          1   
4        Zaglossus_bruijnii  1000005          1   

                       mainCommonName  \
0                            Platypus   
1                Short-beaked Echidna   
2  Attenborough's Long-beaked Echidna   
3         Eastern Long-beaked Echidna   
4         Western Long-beaked Echidna   

                                    otherCommonNames    subclass  \
0                               Duck-billed Platypus  Yinotheria   
1  Australian Echidna|Short-nosed Echidna|Spiny A...  Yinotheria   
2  Attenborough's Echidna|Attenborough's Long-nos...  Yinotheria   
3                       Barton's Long-beaked Echidna  Yinotheria   
4  Long-beaked Echidna|Long-nosed Echidna|Long-no...  Yinotheria   

         infraclass magnorder superorder        order  ...  \
0  Australosphenida       NaN        NaN  MONOTREMATA  ...   
1  Australosphenida       NaN        NaN  MONOTREMATA  ...   
2  Australosphenida       NaN        NaN  MONOTREMATA  ...   
3  Australosphenida       NaN        NaN  MONOTREMATA  ...   
4  Australosphenida       NaN        NaN  MONOTREMATA  ...   

    biogeographicRealm iucnStatus extinct domestic flagged  \
0  Australasia/Oceania         NT       0        0       0   
1  Australasia/Oceania         LC       0        0       0   
2  Australasia/Oceania         CR       0        0       0   
3  Australasia/Oceania         VU       0        0       0   
4  Australasia/Oceania         CR       0        0       0   

                CMW_sciName diffSinceCMW MSW3_matchtype  \
0  Ornithorhynchus_anatinus            0  sciname match   
1    Tachyglossus_aculeatus            0  sciname match   
2   Zaglossus_attenboroughi            0  sciname match   
3         Zaglossus_bartoni            0  sciname match   
4        Zaglossus_bruijnii            0  sciname match   

               MSW3_sciName diffSinceMSW3  
0  Ornithorhynchus_anatinus             0  
1    Tachyglossus_aculeatus             0  
2   Zaglossus_attenboroughi             0  
3         Zaglossus_bartoni             0  
4         Zaglossus_bruijni             0  

[5 rows x 49 columns]

In [5]:
df_main.shape

(6615, 49)

In [10]:
df_main.columns

Index(['sciName', 'id', 'phylosort', 'mainCommonName', 'otherCommonNames',
       'subclass', 'infraclass', 'magnorder', 'superorder', 'order',
       'suborder', 'infraorder', 'parvorder', 'superfamily', 'family',
       'subfamily', 'tribe', 'genus', 'subgenus', 'specificEpithet',
       'authoritySpeciesAuthor', 'authoritySpeciesYear',
       'authorityParentheses', 'originalNameCombination',
       'authoritySpeciesCitation', 'authoritySpeciesLink', 'holotypeVoucher',
       'holotypeVoucherURIs', 'typeLocality', 'typeLocalityLatitude',
       'typeLocalityLongitude', 'nominalNames', 'taxonomyNotes',
       'taxonomyNotesCitation', 'distributionNotes',
       'distributionNotesCitation', 'subregionDistribution',
       'countryDistribution', 'continentDistribution', 'biogeographicRealm',
       'iucnStatus', 'extinct', 'domestic', 'flagged', 'CMW_sciName',
       'diffSinceCMW', 'MSW3_matchtype', 'MSW3_sciName', 'diffSinceMSW3'],
      dtype='object')

In [7]:
len(df_main['mainCommonName'].unique())

6615

In [9]:
df_main['subclass'].unique()

array(['Yinotheria', 'Theria'], dtype=object)

In [12]:
df_main['infraclass'].unique()

array(['Australosphenida', 'Marsupialia', 'Placentalia'], dtype=object)

In [15]:
len(df_main['countryDistribution'].unique())

2429

In [16]:
df_main['order'].unique()

array(['MONOTREMATA', 'PAUCITUBERCULATA', 'DIDELPHIMORPHIA',
       'MICROBIOTHERIA', 'NOTORYCTEMORPHIA', 'DASYUROMORPHIA',
       'PERAMELEMORPHIA', 'DIPROTODONTIA', 'CINGULATA', 'PILOSA',
       'MACROSCELIDEA', 'AFROSORICIDA', 'TUBULIDENTATA', 'PROBOSCIDEA',
       'HYRACOIDEA', 'SIRENIA', 'DERMOPTERA', 'SCANDENTIA', 'PRIMATES',
       'LAGOMORPHA', 'RODENTIA', 'EULIPOTYPHLA', 'CHIROPTERA',
       'PHOLIDOTA', 'CARNIVORA', 'PERISSODACTYLA', 'ARTIODACTYLA'],
      dtype=object)

In [17]:
len(df_main['order'].unique())

27

In [18]:
df_main['family'].unique()

array(['ORNITHORHYNCHIDAE', 'TACHYGLOSSIDAE', 'CAENOLESTIDAE',
       'DIDELPHIDAE', 'MICROBIOTHERIIDAE', 'NOTORYCTIDAE', 'DASYURIDAE',
       'MYRMECOBIIDAE', 'THYLACINIDAE', 'CHAEROPODIDAE', 'PERAMELIDAE',
       'THYLACOMYIDAE', 'HYPSIPRYMNODONTIDAE', 'MACROPODIDAE',
       'POTOROIDAE', 'ACROBATIDAE', 'PETAURIDAE', 'PSEUDOCHEIRIDAE',
       'TARSIPEDIDAE', 'BURRAMYIDAE', 'PHALANGERIDAE', 'PHASCOLARCTIDAE',
       'VOMBATIDAE', 'CHLAMYPHORIDAE', 'DASYPODIDAE', 'BRADYPODIDAE',
       'CHOLOEPODIDAE', 'CYCLOPEDIDAE', 'MYRMECOPHAGIDAE',
       'MACROSCELIDIDAE', 'CHRYSOCHLORIDAE', 'POTAMOGALIDAE',
       'TENRECIDAE', 'ORYCTEROPODIDAE', 'ELEPHANTIDAE', 'PROCAVIIDAE',
       'DUGONGIDAE', 'TRICHECHIDAE', 'CYNOCEPHALIDAE', 'PTILOCERCIDAE',
       'TUPAIIDAE', 'CERCOPITHECIDAE', 'HOMINIDAE', 'HYLOBATIDAE',
       'AOTIDAE', 'ATELIDAE', 'CALLITRICHIDAE', 'CEBIDAE', 'PITHECIIDAE',
       'TARSIIDAE', 'DAUBENTONIIDAE', 'CHEIROGALEIDAE',
       'ARCHAEOLEMURIDAE', 'INDRIIDAE', 'LEMURIDAE', 'L

In [19]:
len(df_main['family'].unique())

167

In [21]:
df_main['superorder'].unique()

array([nan, 'Ameridelphia', 'Australidelphia', 'Xenarthra', 'Afrotheria',
       'Euarchontoglires', 'Laurasiatheria'], dtype=object)

In [22]:
df_main['genus'].unique()

array(['Ornithorhynchus', 'Tachyglossus', 'Zaglossus', ..., 'Mesoplodon',
       'Tasmacetus', 'Ziphius'], dtype=object)

In [23]:
len(df_main['genus'].unique())

1347

In [24]:
len(df_main['mainCommonName'])

6615

This is the tree: subclass (2 values) -> infraclass (3 values) -> order (27 values) -> family (167 values) -> genus (1347 values) -> mainCommonName (6615 values)